src/transport_performance/gtfs/validation.py

```diff
@@ -601,3 +655,803 @@ def get_route_modes(self):
+                join_vars = (
+                    message.split("(")[1]
+                    .replace(")", "")
+                    .replace(" ", "")
+                    .split(",")
+                )
+                drop_cols = [
+                    col
+                    for col in GTFS_UNNEEDED_COLUMNS[table]
+                    if col not in join_vars
+                ]
+                filtered_tbl = table_map[table].copy().drop(drop_cols, axis=1)
+                impacted_rows = self._create_extended_repeated_pair_table(
+                    table=filtered_tbl,
+                    join_vars=join_vars,
+                    original_rows=rows,
+                )
+                base_columns = [
+                    item
+                    for item in list(filtered_tbl.columns)
+                    if item not in join_vars
+                ]
+                duplicate_counts = {}
+                for col in base_columns:
+                    duplicate_counts[col] = impacted_rows[
+                        impacted_rows[f"{col}_original"]
+                        == impacted_rows[f"{col}_duplicate"]
+                    ].shape[0]
+            else:
+                impacted_rows = table_map[table].copy().iloc[rows]
+
+            # create the html to display the impacted rows (clean possibly)
+            table_html = f"""
+            <head>
+                <link rel="stylesheet" href="styles.css">
+            </head>
+            <body>
+            <h1 style="font-family: 'Poppins', sans-serif;margin: 10px;">
+                Table:{table}<br>Message: {message}<br>
+                Type: <span style="color:{'red' if msg_type == 'error' else
+                        'orange'};font-family: 'Poppins', sans-serif;">
+                        {msg_type}</span>
+            </h1>"""
+
+            # Add additional information for repeated pairs
+            # to the HTML report
+            try:
+                for counter, var in enumerate(duplicate_counts):
+                    if counter == 0:
+                        table_html = (
+                            table_html
+                            + """<br><div
+                        style="margin:10px;">
+                        <span style="font-weight:bold;font-size:large">
+                        Duplicate Counts</span>"""
+                        )
+                    table_html = table_html + (
+                        f"""
+                    <div style="display: block;">
+                            <dd style="font-weight: bold;
+                                margin-inline-start: 0;
+                                display: inline-block;
+                                min-width: 160px;">{var}: </dd>
+                            <dt style="display: inline-block;">
+                            {duplicate_counts[var]}</dt>
+                    </div>"""
+                    )
+                table_html = table_html + "</div>"
```

Comment:
The last block with no coverage in this diff. Would this need a check of the table HTML structure? If it's challenging to hit because of the logic above, it could be an idea to move this to a separate function where it would be simpler to feed in dummy `duplicate_counts` objects.

Reply:
As far as I can tell, there is no coverage as there are no repeated pairs in the test fixture.

Reply:
agreed, closing out as linked to #133